Meta refresh editorial updates (#1831)
auto-wcag/auto-wcag: _rules/meta-refresh-no-delay-bc659a.md

## Expectation

> For each test target, running the [shared declarative refresh steps][] results in _time_ being 0 or greater than 72000 (20 hours).

Reviewer: I don't thing https://github.com/act-rules/act-rules.github.io/pull/1831#discussion_r860798971 has been resolved.
The shared declarative refresh steps expect some inputs which I do think are ambiguous here (the document could be the target's document or the target's top-level browsing context document, for example). 

It makes the new expectation ambiguous.

Author: DO you have a suggestion?

Reviewer: Not really… The previous formulation was saying was are the arguments to the algorithm. These changes are requested by TF, I am not part of it so I do not know the discussions and the reasons why TF wants that to be changed…
As far as I am concerned, the previous formulation was OK 🤷

Reviewer 2: I made this suggestion, since I thought that the previous expectation was just duplicating information that was already available inside of the input aspects and the applicability (which I still kind of believe). I didn't think about the target's document vs the target's top-level browsing context, though I'm not sure this would impact the rule significantly in practice. And from your comment, I do see that the previous definition was using the same format given by the `shared declarative refresh steps` algorithm. At this point I could go either way on it.

> For each test target, running the [shared declarative refresh steps][] results in _time_ being 0 or greater than 72000 (20 hours).

Reviewer 3: ```suggestion
For each target, the _time_ from the content [attribute value][] is between 0 and 72000 (20 hours). To determine the _time_, run the [shared declarative refresh steps][] on the `meta` element as described in the [HTML refresh state](https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-http-equiv-refresh).
```

> For each test target, running the [shared declarative refresh steps][], refresh steps results in _time_ being 0 or greater than 72000 (20 hours).

Reviewer 3: ```suggestion
For each test target, running the [shared declarative refresh steps][] results in _time_ being 0 or greater than 72000 (20 hours).
```

> For each test target, running the [shared declarative refresh steps][], refresh steps results in _time_ being 0 or greater than 72000 (20 hours).

Reviewer: I'm not sure about removing the paramters from the expectation.
The [shared declarative refresh steps](https://html.spec.whatwg.org/multipage/semantics.html#shared-declarative-refresh-steps) state:
> The shared declarative refresh steps, given a [Document](https://html.spec.whatwg.org/multipage/dom.html#document) object _document_, string _input_, and optionally a [meta](https://html.spec.whatwg.org/multipage/semantics.html#the-meta-element) element _meta_, are as follows

I'm not sure it is unambiguous enough to call on them without specifying _document_, _input_, and _meta_. Especially when we do then refer to _time_, which is declared in that algorithm (to store the result of the computation).

(typically, _document_ could be the target's document or it top-level browsing context document)

> For each target, the _time_ from the content [attribute value][] is between 0 and 72000 (20 hours). To determine the _time_, run the [shared declarative refresh steps][] on the `meta` element as described in the [HTML refresh state](https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-http-equiv-refresh).

Reviewer: I do like this formulation. I still think we should specify what are the inputs for the algorithm:
- "a `meta` element _meta_": that is probably unambiguous enough given that the rule targets a `meta` element.
- "[a] string _input_": it is not clear that this string should be the value of the `content` attribute of the target.
- "a `Document` object _document_": is also a bit ambiguous (is it the target's owner document, or its top-level browsing context document?)

Reviewer 3: I disagree. By saying "as described in the HTML refresh state", we're telling you how to get the input needed to run the steps. It's easier to read, and if the input ever changes for whatever reason we have it covered.

Reviewer: Ah, yes, I see your point.
It's a bit buried in the third step of the "Will declaratively refresh", but I guess that's worth the simplification 👍

## Background

> This rule is designed specifically for [2.2.1 Timing Adjustable][sc221], which can be satisfied if the time limit is over 20 hours long. All pages that fail this because of a "refresh" `meta` element also do not satisfy [3.2.5 Change on Request][sc325]. In order to adequately test the [expectation](#expectation), some of the passed examples do not satisfy [3.2.5 Change on Request][sc325]. Note: Because a refresh with a timing of 0 is a redirect, it is not party to this rule. However, it is still advised that you not do this. 

Reviewer 3: ```suggestion
This rule is designed specifically for [2.2.1 Timing Adjustable][sc221], which can be satisfied if the time limit is over 20 hours long. All pages that fail this because of a "refresh" `meta` element also do not satisfy [3.2.5 Change on Request][sc325]. In order to adequately test the [expectation](#expectation), some of the passed examples do not satisfy [3.2.5 Change on Request][sc325]. Note: Because a refresh with a timing of 0 is a redirect, it is not party to this rule. Since this can cause rapid screen flashes it is strongly recommend to avoid this.
```

> This rule is designed specifically for [2.2.1 Timing Adjustable][sc221], which can be satisfied if the time limit is over 20 hours long. All pages that fail this because of a "refresh" `meta` element also do not satisfy [3.2.5 Change on Request][sc325]. In order to adequately test the [expectation](#expectation), some of the passed examples do not satisfy [3.2.5 Change on Request][sc325]. Note: Because a refresh with a timing of 0 is a redirect, it is not party to this rule. However, it is still advised that you not do this. 

Reviewer: ```suggestion
This rule is designed specifically for [2.2.1 Timing Adjustable][sc221], which can be satisfied if the time limit is over 20 hours long. All pages that fail this because of a "refresh" `meta` element also do not satisfy [3.2.5 Change on Request][sc325]. In order to adequately test the [expectation](#expectation), some of the passed examples do not satisfy [3.2.5 Change on Request][sc325]. Note: Because a refresh with a timing of 0 is a redirect, it is not a failure of [2.2.1 Timing Adjustable][sc221] and thus not tested by this rule. Since this can cause rapid screen flashes it is strongly recommend to avoid this.
```
(including @WilcoFiers suggestion in mine to make it easier to accept changes)

> This rule is designed specifically for [2.2.1 Timing Adjustable][sc221], which can be satisfied if the time limit is over 20 hours long. All pages that fail this because of a "refresh" `meta` element also do not satisfy [3.2.5 Change on Request][sc325]. In order to adequately test the [expectation](#expectation), some of the passed examples do not satisfy [3.2.5 Change on Request][sc325]. Note: Because a refresh with a timing of 0 is a redirect, it is not party to this rule. Since this can cause rapid screen flashes it is strongly recommend to avoid this.

Reviewer: ```suggestion
This rule is designed specifically for [2.2.1 Timing Adjustable][sc221], which can be satisfied if the time limit is over 20 hours long. All pages that fail this because of a "refresh" `meta` element also do not satisfy [3.2.5 Change on Request][sc325]. In order to adequately test the [expectation](#expectation), some of the passed examples do not satisfy [3.2.5 Change on Request][sc325]. Note: Because a refresh with a timing of 0 is a redirect, it is not part to this rule. Since this can cause rapid screen flashes it is strongly recommend to avoid this.
```

Author: 'is not party to' is the correct language, but very legalese. I recommend changing it to "it is exempt from this rule"

Reviewer: OH… That kind of party 🥳 🎉 
Indeed, I completely missed the idea and thought it was a typo. I agree with your suggestion.

#### Passed Example 1

> This page redirects to a new page immediately.

Reviewer: There's a bit of  a general discrepancy in the descriptions that sometimes refer to "this page" and sometimes to "this meta element". Given that the rule tangets the `meta` element, I guess we should use "this meta element" or "the meta element on this page".

This is however fairly minor, and descriptions are good as they are.

#### Passed Example 2

> First valid `meta` element on the page redirects to a new page immediately.

Reviewer: ```suggestion
First valid `meta` element on this page redirects to a new page immediately.
```

> First valid `meta` element on this page redirects to a new page immediately.

Reviewer 3: ```suggestion
The first valid `meta` element on this page redirects to a new page immediately.
```

> First valid `<meta http-equiv="refresh">` on the page redirects to a new page immediately.

Reviewer 3: We don't want to put HTML inline anymore. Try just using the node name. If necessary use the attributes as a sentence.

```suggestion
First valid `meta` element on the page redirects to a new page immediately.
```

#### Failed Example 1

> This page efreshes after 30 seconds.

Reviewer 4: ```suggestion
This page refreshes after 30 seconds.
```

```html
<head>
	<meta http-equiv="refresh" content="30" />
</head>
<body>
	<p>This page refreshes after 30 seconds.</p>
</body>
```

#### Failed Example 2

> The page redirects to a new page after 30 seconds.

Reviewer: ```suggestion
This page redirects to a new page after 30 seconds.
```

```html
<head>
	<meta http-equiv="refresh" content="30; URL='https://w3.org'" />
</head>
<body>
	<p>This page redirects afte 30 seconds.</p>
</body>
```

#### Failed Example 3

> The first `<meta http-equiv="refresh">` element on the page is not valid, so the second `<meta http-equiv="refresh">` redirects to a new page after 5 seconds.

Reviewer 3: ```suggestion
The first `meta` element on the page is not valid because it uses colon (":") rather than semicolon (";"). The second `meta` element redirects to a new page after 5 seconds.
```

> The first `meta` element on the page is not valid because it uses colon (":") rather than semicolon (";"). The second `meta` element redirects to a new page after 5 seconds.

Reviewer: ```suggestion
The first `meta` element on this page is not valid because it uses colon (":") rather than semicolon (";"). The second `meta` element redirects to a new page after 5 seconds.
```

```html
<head>
	<meta http-equiv="refresh" content="0: https://w3.org" />
	<meta http-equiv="refresh" content="5; https://w3.org" />
</head>
<body>
	<p>This page refreshes after 5 seconds.</p>
</body>
```

#### Failed Example 4

This page redirects to a new page after exactly 20 hours.

```html
<head>
	<meta http-equiv="refresh" content="72000; https://w3.org" />
</head>
<body>
	<p>This page redirects after exactly 20 hours.</p>
</body>
```

### Inapplicable

#### Inapplicable Example 1

This page will not refresh because it lacks a `content` attribute.

```html
<head>
	<meta http-equiv="refresh" />
</head>
<body>
	<p>This page does not refresh.</p>
</body>
```

#### Inapplicable Example 2

> This page will not refresh because it lacks a `http-equiv="refresh"` attribute.

Reviewer 3: ```suggestion
This page will not refresh because it lacks a `http-equiv` attribute.
```

I'll let you do the rest from here.
